tests/robots/test_robot.py

```diff
@@ -289,6 +291,52 @@ def test_get_configurable_joints_wo_semantics(panda_robot_instance_wo_semantics)
     assert all(matches)
 
 
+def test_semantics_serialization(panda_srdf, panda_urdf):
+    model = RobotModel.from_urdf_file(panda_urdf)
+    semantics = RobotSemantics.from_srdf_file(panda_srdf, model)
+    semantics_string = json_dumps(semantics)
+    semantics2 = json_loads(semantics_string)
+    assert isinstance(semantics, RobotSemantics)
+    assert isinstance(semantics2, RobotSemantics)
+    for disabled_collision in semantics.disabled_collisions:
+        assert disabled_collision in semantics2.disabled_collisions
```

Comment:
here you should assert also main group name and maybe passive joints as also.
Also, I'm surprised the membership check (`in`) works,

Reply:
Main group name assertion added.

I think the membership check is passing the string key around. So that works with checking the key in other_dict.